### ISSUE TITLE
Just once and then never more …

### ISSUE THREAD

**Issue by mcepl** [neffo/bing-wallpaper-gnome-extension#57]
It happened to me twice out of two attempts to use your extension on two different computers both with openSUSE/Tumbleweed (Gnome 3.32.2 from the official openSUSE packages).

First nothing happens after the activation, then I select “Refresh Now” in the extension menu, and the image of the day gets downloaded from the Bing. And that’s it the wallpaper never updates, neither automatically nor when I press “Refresh Now” again (even the second day or after a week, with sufficient delay).

Full output of ``journalctl -xb`` is [attached](https://github.com/neffo/bing-wallpaper-gnome-extension/files/3752618/journalctl-log-2019-10-21.txt.zip).

This is not duplicate of #56, because for me **no image** got updated.

**Comment by neffo** [comment]
Hmm, in the extension preferences (can you open it? is the icon visible on the task bar?) can you please check to see where the images are supposed to be downloaded to? Is the directory writable or does it not exist? Non-English languages aren't well tested by me, so probably some bugs there.

**Comment by mcepl** [comment]
As you can see from the settings and Nautilus

![bing-settings](https://user-images.githubusercontent.com/198999/67696624-7fc54480-f9a7-11e9-9231-655087f254e2.png)

directory exists and it is accessible to me (it was created by my normal user, after all). Is it possible somehow to get some log of its actions when I press "Refresh Now". Journalctl doesn’t show anything.

**Comment by neffo** [comment]
You can, there is a debug option that you can enable:

` dconf write /org/gnome/shell/extensions/bingwallpaper/debug-logging true`

**Comment by mcepl** [comment]
Yup, it looks more interesting:
```
Nov 11 09:29:38 stitny gnome-shell[2521]: BingWallpaper extension: next check in 86400 seconds @ local time [boxed instance wrapper GIName:GLib.DateTime jsobj@0x7fce4686b820 native@0x559428eb2700]
Nov 11 09:29:38 stitny gnome-shell[2521]: BingWallpaper extension: market: cs-CZ
Nov 11 09:29:38 stitny gnome-shell[2521]: BingWallpaper extension: fetching: https://www.bing.com/HPImageArchive.aspx?format=js&idx=0&n=1&mbl=1&mkt=cs-CZ
Nov 11 09:29:38 stitny gnome-shell[2521]: BingWallpaper extension: Recieved 866 bytes
Nov 11 09:29:38 stitny gnome-shell[2521]: BingWallpaper extension: JSON returned (raw):
Nov 11 09:29:38 stitny gnome-shell[2521]: {"market":{"mkt":"en-WW"},"images":[{"startdate":"20191111","fullstartdate":"201911110800","enddate":"20191112","url":"/th?id=OHR.MountHowitt_ROW2216892912_1920x1080.jpg&rf=LaDigue_1920x1080.jpg&pid=hp","urlbase":"/th?id=OHR.MountHowitt_ROW2216892912","copyright":"Wildflowers in the mist on Mount Howitt in Alpine National Park, Victoria, Australia (© Australian Scenics/Photolibrary/Getty Images Plus)","copyrightlink":"javascript:void(0)","title":"","quiz":"/search?q=Bing+homepage+quiz&filters=WQOskey:%22HPQuiz_20191111_MountHowitt%22&FORM=HPQUIZ","wp":true,"hsh":"14f854fed6539fc04c4c1e50145ec180","drk":1,"top":1,"bot":1,"hs":[]}],"tooltips":{"loading":"Načítání…","previous":"Předchozí obrázek","next":"Další obrázek","walle":"Tento obrázek není dostupný ke stažení jako tapeta.","walls":"Stáhněte si tento obrázek. Můžete ho ale použít jenom jako tapetu."}}
Nov 11 09:29:38 stitny gnome-shell[2521]: BingWallpaper extension: Mismatched market data, Req: cs-CZ != Recv: en-WW)
```

**Comment by neffo** [comment]
Ahh, ok it's related to the mismatched region versus the data you get. Some regions (what Bing calls markets) aren't available outside the geographic region (or GeoIP region). I'll put a fix in place, but in the mean time you can go to the file at:

`$HOME/.local/share/gnome-shell/extensions/BingWallpaper@ineffable-gmail.com/extension.js`

At line 289 or near abouts you'll see this:
```
 if (datamarket != prefmarket) {
          log('Mismatched market data, Req: '+prefmarket +' != Recv: ' + datamarket +')');
          this.title = _("Market not available in your region");
          this.copyright = 'Request: '+prefmarket +' Recv: ' + datamarket;
          this.imageinfolink = '';
          this._setMenuText();
          this._updatePending = false;
          return;
        }
```

Change the last line to:

`// return;`

That is, comment the return out. The extension won't care from that point on. Sorry, it's not obvious to the user what has happened here. You will probably find that some text will be in English, rather than your native language. It's a weird Bing thing.

Next version will have this add behaivour fixed, I promise :)

**Comment by mcepl** [comment]
OK, it helped, restart of Gnome and "Refresh Now" leads to the today’s page (truly spectacular wild flowers from Australia).

**Comment by neffo** [comment]
Reopening this until the commit is live on GNOME extensions page, thanks for helping track down this bug Matěj, much appreciated!

**Comment by mcepl** [comment]
This is not correct. Update does happen, but the lock screen is still the old one. So, no, this bug may have been fixed, but not #56.

**Comment by mcepl** [comment]
Sorry, I was wrong. With the extension from 62f8544 I get even the lock screen updated.
